src/plugins/advanced_settings/public/management_app/settings.tsx

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useState, useCallback } from 'react';
+import useEffectOnce from 'react-use/lib/useEffectOnce';
+import {
+  EuiSpacer,
+  Query,
+  EuiNotificationBadge,
+  EuiTab,
+  EuiTabs,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+} from '@elastic/eui';
+import { ScopedHistory } from '@kbn/core-application-browser';
+import { IUiSettingsClient } from '@kbn/core-ui-settings-browser';
+import { DocLinksStart } from '@kbn/core-doc-links-browser';
+import { ToastsStart } from '@kbn/core-notifications-browser';
+import { ThemeServiceStart } from '@kbn/core-theme-browser';
+import { UiCounterMetricType } from '@kbn/analytics';
+import { url } from '@kbn/kibana-utils-plugin/common';
+import { parse } from 'query-string';
+import { UiSettingsScope } from '@kbn/core-ui-settings-common';
+import { DEFAULT_CATEGORY, fieldSorter, getAriaName, toEditableConfig } from './lib';
+import { parseErrorMsg } from './components/search/search';
+import { AdvancedSettings, QUERY } from './advanced_settings';
+import { ComponentRegistry } from '..';
+import { Search } from './components/search';
+import { FieldSetting } from './types';
+import { i18nTexts } from './i18n_texts';
+
+interface AdvancedSettingsState {
+  footerQueryMatched: boolean;
+  query: Query;
+  filteredSettings: Record<UiSettingsScope, Record<string, FieldSetting[]>>;
+}
+
+export type GroupedSettings = Record<string, FieldSetting[]>;
+
+interface Props {
+  history: ScopedHistory;
+  enableSaving: boolean;
+  uiSettings: IUiSettingsClient;
+  globalUiSettings: IUiSettingsClient;
+  docLinks: DocLinksStart['links'];
+  toasts: ToastsStart;
+  theme: ThemeServiceStart['theme$'];
+  componentRegistry: ComponentRegistry['start'];
+  trackUiMetric?: (metricType: UiCounterMetricType, eventName: string | string[]) => void;
+}
+
+const SPACE_SETTINGS_ID = 'space-settings';
+const GLOBAL_SETTINGS_ID = 'global-settings';
+
+export const Settings = (props: Props) => {
+  const { componentRegistry, history, toasts, uiSettings, globalUiSettings, ...rest } = props;
+
+  const mapConfig = (config: IUiSettingsClient) => {
```

Comment:
NIT: (Optional given it's tech debt and wasn't introduced by this PR): I would extract all static helpers ( `mapConfig` , `mapSettings`, `initCategoryCounts`, `initCategories`) from the FC's body, or even to a dedicated file to increase readability (and testability)